sdk/confidentialledger/azure-confidentialledger/azure/confidentialledger/receiptverification/receipt_verification.py

```diff
@@ -0,0 +1,275 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+"""Algorithm implementation for verifying Azure Confidential Ledger write
+transaction receipts."""
+
+from base64 import b64decode
+from hashlib import sha256
+from typing import Dict, List, Any, cast
+
+from cryptography.x509 import load_pem_x509_certificate, Certificate
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.asymmetric import ec, utils
+from cryptography.hazmat.primitives.serialization import Encoding, PublicFormat
+
+from azure.confidentialledger.receiptverification.models import (
+    LeafComponents,
+    ProofElement,
+    Receipt,
+)
+
+from azure.confidentialledger.receiptverification._utils import (
+    _convert_dict_to_camel_case,
+)
+
+
+def verify_receipt_from_dict(
+    receipt_dict: Dict[str, Any], service_cert_str: str
```

Comment:
Why do we have a separate method for this? Usually in Python we wouldn't have a separate method for a different input type:
```python
def verify_receipt(receipt: Union[Receipt, Dict[str, Any], service_cert: str) -> None
```
(On another note - is there a particular reason that service_cert is suffixed with the type?)

Reply:
The reason for having a separate method was simply because we had to do some form of "pre-processing" if the input is a dictionary and thought it could be cleaner to have two separate methods for this. But I can have a single method and check for the input type to decide whether to do the pre-processing step or not.

No reason for `service_cert_str` to be suffixed with the type, I will remove it. Thanks for the advice!

Reply:
Fixed in https://github.com/Azure/azure-sdk-for-python/pull/27265/commits/28f7642c76da39c07e66c9cfae35a7dff46807f0